src/Integration/Sentry/SentryMiddleware.php

```diff
@@ -26,7 +26,11 @@ public function process(Request $request, HttpKernelInterface $next): \Iterator
             yield $next->handle($request);
         } finally {
             $client = $this->hub->getClient();
-            $client?->flush()->wait(false);
+
+            $result = $client?->flush();
+            if (is_a($result, 'GuzzleHttp\Promise\PromiseInterface')) {
```

Comment:
I would prefer `instanceof`

```suggestion
            if ($result instanceof PromiseInterface) {
```

Reply:
But there will be no PromiseInterface if guzzle is not installed.

Reply:
It's not an issue with PHP, it will simply return `false`. See https://3v4l.org/p288k